messages_macros/src/lib.rs

```diff
@@ -0,0 +1,63 @@
+#![allow(clippy::expect_fun_call)]
+
+mod message_type;
+
+use message_type::message_type_impl;
+use proc_macro::TokenStream;
+use syn::{parse_macro_input, DeriveInput, Error};
+
+/// Derive macro to be used for easier implementation of message type components.
+/// The macro serves as implementation for semver reasoning and parsing of the `@type` field
+/// components (major, minor versions, message kind, etc.) in a message.
+///
+/// The macro can only be derived on *newtype enums*, expecting either
+/// a protocol or a major version encapsulating enum.
+///
+/// The minor versions are represented by the major version enum's variants
+/// and the field encapsulated in the variants are expected to be [`crate::msg_types::MsgKindType<T>`].
```

Comment:
Yeah, I know. Maybe it would be better to just use the type name alone instead of the absolute path. Just wanted to have the type name there to be more descriptive. Will fix.

Reply:
Excluded the path, should be more readable now.